flaps/flaps.go

```diff
@@ -184,6 +188,12 @@ func (f *Client) _sendRequest(ctx context.Context, method, endpoint string, in,
 		}
 	}()
 
+	defer func() {
+		if statusCode != nil {
+			*statusCode = resp.StatusCode
+		}
+	}()
```

Comment:
These lines triggered some eye twitching :))

Why not returning the statuscode (along the error) if we are already changing the function signature? 

I mean, this approach of returning values is from languages that doesn't support returning multiple values and that is not the Go case.

Reply:
Yea that's understandable, I just didn't know how annoying it would be to always return a status code when it's only needed in one function.